### PR TITLE
Prepare for migration

### PR DESCRIPTION
**Pull request by ben-kaufman**
Prepares for V2 migration based on the plan detailed in #93

### CODE REVIEW

File: contracts/HATVaults.sol [hunk 1]
```diff
@@ -949,20 +939,48 @@ contract  HATVaults is Governable, ReentrancyGuard {
...
         poolDepositPause[_pid] = _depositPause;
         emit SetPool(_pid, _allocPoint, _registered, _descriptionHash);
     }
 
+    function setPoolInitialized(uint256 _pid) external onlyGovernance {
+        require(poolInfo.length > _pid, "HVE23");
+       poolInitialized[_pid] = true;
+    } 
+
+    function setShares(
```
Comment (lirona) [inline contracts/HATVaults.sol:953]: should set poolInitialized to true here and not in a separate function, no?

File: contracts/HATVaults.sol [hunk 2]
```diff
@@ -949,20 +939,48 @@ contract  HATVaults is Governable, ReentrancyGuard {
...
-        require(poolInfos.length > _pid, "HVE23");
+        require(poolInfo.length > _pid, "HVE23");
         set(_pid, _allocPoint);
         poolDepositPause[_pid] = _depositPause;
         emit SetPool(_pid, _allocPoint, _registered, _descriptionHash);
     }
 
+    function setPoolInitialized(uint256 _pid) external onlyGovernance {
+        require(poolInfo.length > _pid, "HVE23");
+       poolInitialized[_pid] = true;
```
Comment (jellegerbrandy) [inline contracts/HATVaults.sol:950]: indentation

File: contracts/HATVaults.sol [hunk 3]
```diff
@@ -949,20 +939,48 @@ contract  HATVaults is Governable, ReentrancyGuard {
...
+    external onlyGovernance {
+        require(!poolInitialized[_pid], "HVE38");
+        require(poolInfo.length > _pid, "HVE23");
+        require(_accounts.length == _shares.length, "HVE39");
+        require(_accounts.length == _rewardDebts.length, "HVE39");
+        PoolInfo storage pool = poolInfo[_pid];
+        pool.rewardPerShare = _rewardPerShare;
+        pool.balance = _balance;
+        for (uint256 i = 0; i < _accounts.length; i++) {
+            userInfo[_pid][_accounts[i]] = UserInfo({
```
Comment (jellegerbrandy) [inline contracts/HATVaults.sol:969]: we must also update totalshares here, right?

File: scripts/migration-snapshot.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
+const { network } = require("hardhat");
+const fs = require("fs");
+const CONFIG = require("./addresses.js");
+const HATVaults = JSON.parse(
```
Comment (jellegerbrandy) [inline scripts/migration-snapshot.js:4]: this is the ABI right?

Comment (jellegerbrandy) [reply scripts/migration-snapshot.js:4]: mb name the file in a more clearer way (like hatvaultsv1ABI.json :)

**Review by jellegerbrandy — CHANGES_REQUESTED** [review]

File: contracts/HATVaults.sol (hunk 3 above)
Comment (ben-kaufman) [reply contracts/HATVaults.sol:969]: Fixed

File: contracts/HATVaults.sol (hunk 2 above)
Comment (ben-kaufman) [reply contracts/HATVaults.sol]: Fixed

**Review by jellegerbrandy — APPROVED** [review]

**Pull request merged**